src/autocomplete_test.js

```diff
@@ -462,12 +462,43 @@ module.exports = {
         user.type("Ctrl-Space");
         assert.equal(editor.completer.popup.isOpen, true);
         assert.equal(editor.completer.popup.data[0].caption, emptyMessageText);
+        assert.ok(editor.completer.popup.renderer.container.classList.contains("ace_empty-message"));
 
         user.type("Return");
         assert.equal(editor.completer.popup.isOpen, false);
 
         done();
     },
+    "test: no empty message class if suggestions available": function(done) {
+        var editor = initEditor("");
+        var emptyMessageText = "No suggestions.";
+        var autocomplete = Autocomplete.for(editor);
+        autocomplete.emptyMessage = () => emptyMessageText;
+
+        editor.completers = [
+            {
+                getCompletions: function (editor, session, pos, prefix, callback) {
+                    var completions = [
+                        {
+                            caption: "append",
+                            value: "append"
+                        }
+                    ];
+                    callback(null, completions);
+                }
+            }
+        ];
+
+        user.type("app");
+
+        // Open autocompletion via key-binding and verify no empty message class
+        user.type("Ctrl-Space");
+        assert.equal(editor.completer.popup.isOpen, true);
+        assert.equal(editor.completer.popup.data[0].caption, "append");
+        assert.ok(!editor.completer.popup.renderer.container.classList.contains("ace_empty-message"));
+
```

Comment:
It would be nice to also test the reverse `empty -> results`

Reply:
so trigger it 1 time without results (check class there) and then trigger with results and check class not there?

Reply:
Improved the test to test that scenario as well